### PR TITLE
create comment component

### PR DESCRIPTION
**Pull request by Fl3xm3ist3r**

### CODE REVIEW

File: client/src/app/shared/comment/comment.component.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  @Input()
+  get splitPositions(): number[] {
+    return this.splitPositionValue;
+  }
+  set splitPositions(value: number[]) {
+    this.splitPositionValue = value;
+    setTimeout(() => this.createComment());
+  }
+
+  constructor(private elementRef: ElementRef) { }
```
Comment (Fl3xm3ist3r) [inline client/src/app/shared/comment/comment.component.ts:42]: try to refactor

File: client/src/app/home/home.component.html [hunk 2]
```diff
@@ -1,3 +1,5 @@
+<app-comment [commentType]='commentType.ALL' contents='{{helloWorldCommentValue}}' [splitPositions]='splitPositionValue'></app-comment>
```
Comment (xgiaenuuu) [inline client/src/app/home/home.component.html:1]: Maybe you'll find a better name for commentType. Its is not the type of the comments its more of a splitting mechanism.

File: client/src/app/home/home.component.ts [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
-
+import { CommentType } from '../shared/comment/comment.component';
+import { helloWorldComment } from 'src/assets/utf-styles/hello_world';
 @Component({
   selector: 'app-home',
   templateUrl: './home.component.html',
   styleUrls: ['./home.component.scss']
 })
 export class HomeComponent implements OnInit {
+  public commentType = CommentType;
```
Comment (xgiaenuuu) [inline client/src/app/home/home.component.ts:10]: As I mentioned above...

File: client/src/app/shared/comment/comment.component.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    tempString = tempString.slice(4);
+    tempString = tempString.slice(0, -3);
+    return tempString;
+  }
+}
+
+export enum CommentType{
+  ALL,
+  LINE,
+}
```
Comment (xgiaenuuu) [inline client/src/app/shared/comment/comment.component.ts:136]: Maybe you can create a service for these functions. We want to try to keep the Components as small as possible. Handling the string should be done outside of the component. The Component is actually just for "Displaying".

**Review by xgiaenuuu — CHANGES_REQUESTED** [review]

File: client/src/app/shared/comment/comment.component.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  get splitPositions(): number[] {
+    return this.splitPositionValue;
+  }
+  set splitPositions(value: number[]) {
+    this.splitPositionValue = value;
+    setTimeout(() => this.createComment());
+  }
+
+  constructor(private elementRef: ElementRef) { }
+  private createComment() {
```
Comment (xgiaenuuu) [inline client/src/app/shared/comment/comment.component.ts:43]: Maybe you can move all those functions to a service. The goal is, to just really have DOM logic in the component and no data handling.

**Review by xgiaenuuu — CHANGES_REQUESTED** [review]

**Review by xgiaenuuu — APPROVED** [review]
:+1:

**Pull request merged**